src/deepsparse/transformers/pipelines/chat.py

```diff
@@ -0,0 +1,228 @@
+        :return: the inputs for the engine
+        """
+        engine_input, context = super().process_inputs(inputs)
+        return self.add_session_ids_to_engine_input(engine_input, inputs), context
+
+    def process_engine_outputs(
+        self, engine_outputs: List[Union[numpy.ndarray, FinishReason, str]], **kwargs
+    ) -> ChatOutput:
+        """
+        Convert the engine outputs to the output schema for the pipeline.
+
+        :param engine_outputs: the outputs from the engine
+        :param kwargs: additional keyword arguments
+        :return: the output schema for the pipeline
+        """
+        session_ids = engine_outputs.pop(-1)
+        # process the engine outputs within the context of TextGenerationPipeline
+        text_generation_output = super().process_engine_outputs(
+            engine_outputs, **kwargs
+        )
+        # create the ChatOutput from the data provided
+        return ChatOutput(**text_generation_output.dict(), session_ids=session_ids)
+
+    def engine_forward(
+        self, engine_inputs: List[numpy.ndarray], context: Dict
+    ) -> Tuple[numpy.ndarray, numpy.ndarray, List[FinishReason], str]:
+        """
+        Wrapper around the engine forward function to add session ids to
+        the returned engine outputs
+
+        :param engine_inputs: the inputs for the engine
+        :param context: the context for the engine
+        :return: the outputs from the engine
+        """
+        session_id = engine_inputs[-1]
+        return *super().engine_forward(engine_inputs, context), session_id
+
+    def get_kv_cache_decoder(self, engine_inputs: List[Any]) -> DecoderKVCache:
+        """
+        Initialize the kv cache decoder for the inference.
+        If the storage kv cache does not have a kv cache decoder,
+        create a new one and store it in the storage kv cache.
+
+        :param engine_inputs: the inputs for the engine
+        :return: the initialized kv cache decoder
+        """
+        session_id = engine_inputs[-1]
+        kv_cache = self.storage_kv_cache.get(session_id)
+        if kv_cache is None:
+            kv_cache = super().get_kv_cache_decoder(engine_inputs)
+            self.storage_kv_cache[session_id] = kv_cache
+
+        return kv_cache
+
+    def add_session_ids_to_engine_input(
+        self, engine_input: List[numpy.ndarray], inputs: ChatInput
+    ):
+        """
+        Create the session ids for the input sequences (if not provided)
+        and add them to the engine input.
+
+        :param engine_input: the inputs for the engine
+        :param inputs: the input schema for the pipeline
+
+        :return: the engine input with the session ids
+        """
+        session_ids = inputs.session_ids
+        if session_ids is None:
+            # session_ids is None, so we need to generate
+            # a session id for each input sequence
+            # TODO: Talk to Dipika whether this aligns with the
```

Comment:
I don't think this will - we'll need to do a bit more planning here - cc @dsikka